subsys/bluetooth/host/iso.c

```diff
@@ -729,6 +724,61 @@ void bt_iso_recv(struct bt_conn *iso, struct net_buf *buf, uint8_t flags)
 }
 #endif /* CONFIG_BT_ISO_RX */
 
+static bool iso_has_data(struct bt_conn *conn)
+{
+#if defined(CONFIG_BT_ISO_TX)
+	return !k_fifo_is_empty(&conn->iso.txq);
+#else
+	return false;
+#endif
+}
+
+static struct net_buf *iso_data_pull(struct bt_conn *conn, size_t amount)
+{
+#if defined(CONFIG_BT_ISO_TX)
```

Comment:
I've noticed that you tend to guard the content of the function rather than the declaration of the function. 

Is there some advantage of that I'm not seeing? 

i.e. you are opting for 

```c
void foo(void)
{
#if CONFIG_FOO
    whatever
#endif
}

void bar(void)
{
    foo();
}
```

rather than

```c
#if CONFIG_FOO
void foo(void)
{
    whatever
}
#endif

void bar(void)
{
#if CONFIG_FOO
        foo();
#endif
}
```

Reply:
Simply counting the number of ifdefs there seems to be an advantage to the first approach :)

Reply:
Indeed. 

It has the disadvantage that you can call foo() without any effect, whereas the latter version would give you a compile-time error. This is mostly true for `void` functions, which isn't the case where where the function simply returns `NULL` if not enabled.

Reply:
Yes, the reason I do that is that C is not an ergonomic language. An ideal compiler _could_ figure it out, but it doesn't.

`#ifdef`s in the function logic confuse me. So I use `if (IS_ENABLED(x))` to keep proper indentation. 

That gives the weird ifdeffing-out the whole body thing.
Another pro for that weird solution is that unconfigured editors don't get confused when jumping-to-definition (e.g. editor has to know what config you're building if you're jumping around in `id.c`).